beacon-chain/core/helpers/weak_subjectivity.go

```diff
@@ -202,3 +202,14 @@ func ParseWeakSubjectivityInputString(wsCheckpointString string) (*v1alpha1.Chec
 		Root:  bRoot,
 	}, nil
 }
+
+// MinEpochsForBlockRequests computes the number of epochs of block history that we need to maintain,
+// relative to the current epoch, per the p2p specs. This is used to compute the slot where backfill is complete.
```

Comment:
We'll do a follow up feature to add a flag like `--backfill-to-epoch` that will allow the user to specify an earlier backfill target: `--backfill-to-epoch=0` to go all the way to genesis. Actually this reminds me that I need to file a separate issue for this, currently we only have https://github.com/prysmaticlabs/prysm/issues/13003 which is related but doesn't cover it. Just added this one https://github.com/prysmaticlabs/prysm/issues/13031

Reply:
While this is the minimum by the spec, shouldn't we have this value by default ?
```
backfill-to-epoch=0
```
Reasoning is that on node default's all of the beacon history is always maintained. If a user does not care about it, they can then just give a recent epoch

Reply:
spec text:
```Clients MUST keep a record of signed blocks seen on the epoch range [max(GENESIS_EPOCH, current_epoch - MIN_EPOCHS_FOR_BLOCK_REQUESTS), current_epoch] where current_epoch is defined by the current wall-clock time, and clients MUST support serving requests of blocks on this range.```

Reply:
My stance is that the main purpose of beacon nodes is to participate in the consensus protocol. Following from the principals of finalization and weak subjectivity, most nodes are really only interested in blocks forward from the beginning of the weak subjectivity period. Older is history is not useful for participating in consensus, it is only interesting for archival purposes. So our default behavior should not be to download older history that is not useful to most of the network.

I am open to having my mind changed on this issue :)

Reply:
Rather than participating in consensus, having the ability to download and serve historical blocks should be an expected duty of a normal node. There is no alternate protocol to serve historical blocks from the consensus layer network. If all nodes simply stop saving historical blocks, you would end up with very few peers who would have full chain histories. It is the same reason execution layer nodes also save all historical blocks even when snap syncing.